DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -313,35 +316,46 @@ private void CreateDelegateAccountForAdmin(AdminRegistrationModel registrationMo
                 registrationModel.ProfessionalRegistrationNumber
             );
 
-            var candidateNumberOrErrorCode =
-                registrationDataService.RegisterNewUserAndDelegateAccount(delegateRegistrationModel);
-            var failureIfAny = DelegateCreationError.FromStoredProcedureErrorCode(candidateNumberOrErrorCode);
-            if (failureIfAny != null)
+            try
             {
-                logger.LogError(
-                    $"Delegate account could not be created (error code: {candidateNumberOrErrorCode}) with email address: {registrationModel.PrimaryEmail}"
+                var candidateNumber =
+                    registrationDataService.RegisterNewUserAndDelegateAccount(delegateRegistrationModel);
+                passwordDataService.SetPasswordByCandidateNumber(
+                    candidateNumber,
+                    delegateRegistrationModel.PasswordHash!
                 );
 
-                throw new DelegateCreationFailedException(failureIfAny);
-            }
-
-            passwordDataService.SetPasswordByCandidateNumber(
-                candidateNumberOrErrorCode,
-                delegateRegistrationModel.PasswordHash!
-            );
+                // We know this will give us a non-null user.
+                // If the delegate hadn't successfully been added we would have errored out of this method earlier.

```

Comment:
I think it means that since registrationDataService.RegisterNewUserAndDelegateAccount and             passwordDataService.SetPasswordByCandidateNumber are called in this same method immediately before this, we would have hit an error in one of those already if the user was null